neevaco/neeva-android
# Don't automatically set another tab as active when closing a tab from the switcher

Instead of requiring a tab to be made active after another tab is closed by the user in the tab switcher, we now allow there to be no active tab set.

* Remove code in various places that updates the currently active tab to anything else in the list.

* Prevent the user from leaving the tab grid if the currently active tab is being closed.

* Update the tests

This doesn't fix the pre-existing issue where the user can have no tabs, start creating a lazy tab, then hit back to get out of it and end up on an about:blank page: (https://github.com/neevaco/neeva-android/issues/1072)

## Checklists

### Pointers to help with review

### How was this tested?

### Manual test cases
Hitting back while the active tab is closing (also added as an automated test):
1. Have at least two tabs open in the tab switcher
2. Close the active tab.  Note that there is no active tab selected
3. Immediately hit back before the snackbar disappears
4. See that you're sent out of the app

Hitting back while the active tab has finished closing (also added as an automated test):
1. Have at least two tabs open in the tab switcher
2. Close the active tab.  Note that there is no active tab selected
3. Wait for the snackbar to disappear.  Note that there is still no active tab selected
4. Hit the system back button
5. See that you're sent out of the app

### Screenshots and screen recordings

https://user-images.githubusercontent.com/96092902/197293154-480c20b5-28c4-428c-8e2f-81ea2428ed3f.mov


## Issues addressed
* Fixes https://github.com/neevaco/neeva-android/issues/990
